app/gui/view/debug_scene/cloud-dashboard/src/projects_table.rs

```diff
@@ -0,0 +1,698 @@
+        if idx >= self.projects.len() {
+            warn!(
+                "Attempted to display entry at index {idx}, but we only have data up to index {}.",
+                self.projects.len()
+            );
+            return None;
+        }
+        Some(idx)
+    }
+
+    fn resize_grid_to_shape(&self, shape: &ensogl::display::scene::Shape) -> Vector2<f32> {
+        let screen_size = Vector2::from(shape);
+        let margin = Vector2::new(HORIZONTAL_MARGIN, VERTICAL_MARGIN);
+        let size = screen_size - margin;
+        size
+    }
+
+    fn reposition_grid_to_shape(&self, shape: &ensogl::display::scene::Shape) {
+        let screen_size = Vector2::from(shape);
+        let screen_size_halved = screen_size / 2.0;
+        let viewport_min_y = -screen_size_halved.y + BOTTOM_MARGIN;
+        let viewport_min_x = -screen_size_halved.x + LEFT_MARGIN;
+        let grid_height = screen_size.y * GRID_HEIGHT_RATIO;
+        let grid_min_x = viewport_min_x;
+        let grid_max_y = viewport_min_y + grid_height;
+        let position = Vector2::new(grid_min_x, grid_max_y);
+        self.projects_table.set_xy(position);
+    }
+
+    fn refit_entries_to_shape(&self, shape: &ensogl::display::scene::Shape) -> Vector2<f32> {
+        let width = shape.width / Columns::LEN as f32;
+        let size = Vector2(width, ENTRY_HEIGHT);
+        size
+    }
+
+    /// Returns the information about what section the requested visible entry is in.
+    ///
+    /// The grid view wants to know what to display as a header of the top of the viewport in the
+    /// current column. Therefore, it asks about information about the section the topmost visible
+    /// entry is in.
+    ///
+    /// `section_info_needed` assumes that there may be more than one section, therefore it needs to
+    /// know the span of rows belonging to the returned section (so it knows when to ask about
+    /// another one).
+    ///
+    /// Since we use separate tables to display separate types of data in the dashboard, this table
+    /// only ever has one section. So it should always return the range `1..=self.rows()`.  However
+    /// since we don't have easy access to the number of projects in the table, we just return
+    /// [`usize::MAX`], which is provides what we want because we always want all rows to be part of
+    /// the one section.
+    fn position_to_requested_section(&self, position: Position) -> (Range<Row>, Col, EntryModel) {
+        /// The first row in the section the requested visible entry is in.
+        ///
+        /// This is always `0` because we only have one section, so the first row is the first
+        /// visible one in the table.
+        const SECTION_START: usize = 0;
+        /// The last row in the section the requested visible entry is in.
+        ///
+        /// This is always [`usize::MAX`] because we only have one section, so all rows are in the
+        /// current section.
+        const SECTION_END: usize = usize::MAX;
```

Comment:
It does *seem* to work with `usize::MAX`. Uou are correct that `self.rows()` would be the better solution. It's what I tried initially, but had issues with referencing that function from the `frp::extend!`. However since this code is now in a method on `Model` this doesn't seem to be an issue.

Updated to use `self.rows()` and call `reset_entries` upon receiving the projects list.